tests/unit/test_cli.py

```diff
@@ -99,17 +99,33 @@ def test_main(cli_runner: CliRunner, action: str):
     assert f"Usage: main {action}" in result.output
 
 
-def test_initialize(monkeypatch: pytest.MonkeyPatch, cli_runner: CliRunner):
+@pytest.mark.parametrize(
+    "flags",
+    [
+        pytest.param([], id="No flags (chroot builder)"),
+        pytest.param(
+            ["--experimental-external", "true", "--cloud-name", "hello"],
+            id="External flags (openstack builder)",
+        ),
+    ],
+)
+def test_initialize(monkeypatch: pytest.MonkeyPatch, cli_runner: CliRunner, flags: list[str]):
     """
     arrange: given a monkeypatched builder.initialize function.
     act: when cli init is invoked.
     assert: monkeypatched function is called.
     """
-    monkeypatch.setattr(cli.builder, "initialize", (mock_func := MagicMock()))
+    monkeypatch.setattr(cli.builder, "initialize", (mock_builder_init_func := MagicMock()))
+    monkeypatch.setattr(
+        cli.openstack_builder, "initialize", (mock_openstack_init_func := MagicMock())
+    )
 
-    cli_runner.invoke(main, args=["init"])
+    cli_runner.invoke(main, args=["init", *flags])
 
-    mock_func.assert_called()
+    if not flags:
+        mock_builder_init_func.assert_called()
```

Comment:
Consider using `assert_called_with` instead to make sure the args are ok. Same for the rest of the tests

Reply:
Some of them are internal functions that are already magic-mocked and just requires that they are called. I've updated the ones that requires checking the inputs, thank you!